edgeless_telemetry/src/elasticsearch_api.rs

```diff
@@ -0,0 +1,203 @@
+    Ok(())
+}
+
+//static counter for indexing identification
+fn increment_counter() -> String {
+    unsafe {
+        COUNTER += 1;
+        COUNTER.to_string()
+    }
+}
+
+/// Writes data to an Elasticsearch index.
+/// # Arguments
+/// * `client_result` - the Elasticsearch client
+/// * `data` - The data to be written to the index.
+/// * `index_flag` - which index to write to
+/// # Returns
+/// A Result indicating success or failure.
+
+pub async fn es_write_to_index(
+    client_result: &Result<Elasticsearch, Box<dyn std::error::Error>>,
+    data: Value,
+    index_flag: bool,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let client = match client_result {
+        Ok(client) => client,
+        Err(err) => return Err(err.to_string().into()), // Convert error to String
+    };
+    let id = increment_counter();
+    //send data to ES endpoint via POST
+    let index_response = client
+        .index(IndexParts::IndexId(get_index_name(index_flag), &id))
+        .body(data)
+        .send()
+        .await?;
+
+    println!("Response Status Code: {}", index_response.status_code());
+    Ok(())
+}
+
+pub fn get_current_timestamp() -> DateTime<Utc> {
+    Utc::now()
+}
+
+/// Retrieves data from an Elasticsearch index.
+/// # Arguments
+/// * `client_result` - the Elasticsearch client
+/// * `index_flag` - which index to read   
+/// # Returns
+/// A Result containing a vector of JSON values representing the contents of the index
+pub async fn es_read_from_index(
+    client_result: &Result<Elasticsearch, Box<dyn std::error::Error>>,
+    index_flag: bool,
+) -> Result<Vec<Value>, Box<dyn std::error::Error>> {
+    let client = match client_result {
+        Ok(client) => client,
+        Err(err) => return Err(err.to_string().into()), // Convert error to String
+    };
+
+    let index = get_index_name(index_flag);
+    println!("Contents of index {}", index);
+
+    //fetch data from ES endpoint with index via GET
+    let search_response = client
+        .search(SearchParts::Index(&[index]))
+        .body(json!({
+            "query": {
+                "match_all": {}
+            }
+        }))
+        .send()
+        .await?;
+    if search_response.status_code().is_success() {
+        let hits = search_response.json::<Value>().await?["hits"]["hits"]
+            .as_array()
+            .map_or_else(Vec::new, |hits| hits.clone());
+        let contents: Vec<Value> = hits.iter().map(|hit| hit["_source"].clone()).collect();
+        Ok(contents)
+    } else {
+        Err("Failed to retrieve index contents".into())
+    }
+}
+
+/// Determines the index name based on the flag indicating whether to use the resources index.
+/// # Arguments
+/// * `use_resources_index` - A boolean flag indicating whether to use the resources index (`true`) or the runtime index (`false`).
+/// # Returns
+/// * Returns a string slice (`&'static str`) representing the index name.
+fn get_index_name(use_resources_index: bool) -> &'static str {
```

Comment:
Use an enum instead of a bool. It was not clear what the true/false mean in the code above.